src/coreclr/debug/daccess/request.cpp

```diff
@@ -2610,6 +2610,37 @@ ClrDataAccess::GetPEFileBase(CLRDATA_ADDRESS moduleAddr, CLRDATA_ADDRESS *base)
 
     SOSDacEnter();
 
+    if (m_cdacSos != NULL)
+    {
+        hr = m_cdacSos->GetPEFileBase(moduleAddr, base);
+        if (FAILED(hr))
+        {
+            hr = GetPEFileBaseImpl(moduleAddr, base);
+        }
+#ifdef _DEBUG
+        else
+        {
+            CLRDATA_ADDRESS baseLocal = 0;
+            HRESULT hrLocal = GetPEFileBaseImpl(moduleAddr, &baseLocal);
+
+            DacAssertsEnabledHolder assertsEnabled;
+            _ASSERTE(hr == hrLocal);
+            _ASSERTE(*base == baseLocal);
```

Comment:
You need to be a little careful here.  if the original call had `base == nullptr`, then the managed impl with return some failure error code, but GetPEFileBaseImpl might succeed since you always pass `&baseLocal`.

what I did elsewhere is something like this:

```c
CLRDATA_ADDRESS baseLocal = 0;
CLRDATA_ADDRESS baseLocalPtr = base ? &baseLocal : nullptr;

HRESULT hrLocal = GetPEFileBaseImpl(moduleAddr, baseLocalPtr);
```

then assert that `*base` and `baseLocal` are the same only if `base` is not null